### ISSUE TITLE
🐛 Wrong error message when using JSX in a non-JSX file

### ISSUE THREAD

**Issue by simon-paris** [biomejs/biome#2085]
### Environment information

```block
CLI:
  Version:                      1.6.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/1.22.18"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. Use JSX in a .js file (no typescript, no JSX)
2. The error message is "Type assertion are a TypeScript only feature."

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAZgAoACkAIAB7AAoAIAAgAHIAZQB0AHUAcgBuACAAPABkAGkAdgAvAD4AOwAKAH0A&typescript=false&jsx=false&script=true

### Expected result

It should print a better error message.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by togami2864** [comment]
The parser is currently misinterpreting JSX and the angle-bracket syntax for type assertions in TypeScript.
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions

I think it might be nice to display the expression error similarly to how swc does.
https://play.swc.rs/?version=1.3.100&code=H4sIAAAAAAAAA0srzUsuyczPU0jT0FSo5lJQKEotKS3KU7BJySzTt7PmqgUA6O8O2iEAAAA%3D&config=H4sIAAAAAAAAA1VPOw6DMAzdc4rIc4e2Ujv0Dj1ElBoUlJ9iIxUh7k4ICS2b38%2FPnoWUMJCGl5zzmEFUiTAdODM0eVbfzABqp0gnExkuTR1okzplCQu17AqwSj1ySdH9envUBNgQCFuics54003%2FnTq4mJDobNysyvcWz42itoILn7GI9ReeIu4XPOFnamXHYjD0bklOI4plBQmBQtEXAQAA

**Comment by cdvillard** [comment]
I'm experiencing the same with a very basic React demo I'm writing. This is specifically involving JSX syntax as it occurs for me in both JS and JSX files. 
![image](https://github.com/biomejs/biome/assets/6250214/86905ca4-0dab-4ee0-b55b-0fecdb090996)

**Comment by ematipico** [comment]
I haven't seen concrete suggestions here. OP suggested something. However,  it isn't actionable.

> It should print a better error message.

What kind of error message would you expect? Suggestions are welcome
